prow/sidecar/run.go

```diff
@@ -108,6 +108,10 @@ func (o Options) Run(ctx context.Context) (int, error) {
 		return 0, fmt.Errorf("could not resolve job spec: %v", err)
 	}
 
+	entries := o.entries()
+	buildLogs := logReaders(entries)
+	metadata := combineMetadata(entries)
```

Comment:
`logReaders` and `combineMetadata` cannot be called this early. They need to be called immediately before `doUpload` so that we don't open and read the files before they've been written.

Reply:
Move these calls to immediately before `doUpload` and `bestEffortUpload` call.